src/python/WMCore/MicroService/MSPileup/DataStructs/MSPileupObj.py

```diff
@@ -114,6 +113,11 @@ def validate(self, pdict=None):
                 self.logger.error(msg)
                 return False, msg
             if key == 'pileupName' or (key == 'customName' and val != ''):
+                if key == 'customName':
```

Comment:
From lexicon defition:
```
import re
DATASET_RE = r'^/[a-zA-Z0-9\-_]{1,99}/[a-zA-Z0-9\.\-_]{1,199}/[A-Z\-]{1,50}$'
pat=re.compile(DATASET_RE)
d="/abc/xyz/PREMIX-V1"
pat.match(d)
```
does not match because numbers are not allowed in data-tier part. In other words if DATASET_RE would be:
```
DATASET_RE = r'^/[a-zA-Z0-9\-_]{1,99}/[a-zA-Z0-9\.\-_]{1,199}/[A-Z0-9\-]{1,50}$'
```
(please pay attention to last part, then the data-tier will match. Therefore, either we need to change code (as I did) or change dataset pattern which I doubt we should do since it may break other parts in CMS universe.